include/dem/distributions.h

```diff
@@ -128,4 +129,46 @@ class UniformDistribution : public Distribution
   double diameter_values;
 };
 
+class HistogramDistribution : public Distribution
+{
+public:
+  /**
+   * @brief The constructor stores the parameters necessary to define the histogram
+   * distribution.
+   * @param d_list
+   * @param d_probabilities
+   * @param number_of_particle_types
+   */
+  HistogramDistribution(const std::vector<double> &d_list,
+                        const std::vector<double> &d_probabilities);
+
+  /**
+   * @brief Carries out the size sampling of each particle inserted at a insertion
+   * time step for the histogram distribution.
+   *
+   * @param particle_number Number of particle inserted at a given insertion time step.
+   * @param particle_type The type of particle being inserted.
+   */
+  void
+  particle_size_sampling(const unsigned int &particle_number) override;
+
+  /**
+   * @brief Return the minimal diameter value from the @diameter_hist_value vector.
+   */
+  double
+  find_min_diameter() override;
+
+  /**
+   * @brief Return the maximal diameter value from the @diameter_hist_value vector.
+   */
+  double
+  find_max_diameter() override;
```

Comment:
I think you can add the **return** description in the brief (just so that it is uniform, even if it is clear in the brief haha)